test/e2e/sourcery.js

```diff
@@ -1,62 +1,108 @@
+        let billOption = "Yes";
+        let taskName = "Chuck Testas";
+        let taskDescription = 'This task is made for learning purposes, during which we will learn test automation';
 
-        browser.end();
+        browser
+        .isVisible(timeLogging.tasksButton, function(result){  //Click the Tasks section
+            if(result.status === c.ELEMENT_FOUND)
+                browser.click(timeLogging.tasksButton);
+        })
+        .assert.containsText(common.pageTitle, pageNameTasks)  //Check if the page is 'Tasks'
+        .isVisible(tasks.createTaskButton, function(result){  //Click the Create Task button
+            if(result.status === c.ELEMENT_FOUND)
+                browser.click(tasks.createTaskButton);
+        })
+        .assert.containsText(common.pageTitle, pageNameCreateTask) //Check if the page is 'Create task'
+        .isVisible(createTask.taskName, function(result){ //Enter task name
+            if(result.status === c.ELEMENT_FOUND)
+                browser.setValue(createTask.taskName, taskName)
+        })
+        .isVisible(createTask.taskDescription, function(result){ //Enter task description
+            if(result.status === c.ELEMENT_FOUND)
+                browser.setValue(createTask.taskDescription, taskDescription);
+        })
+        .isVisible(createTask.taskBill, function(result){ //Click the 'Bill to Client' dropdown
+            if(result.status === c.ELEMENT_FOUND)
+                browser.click(createTask.taskBill);
+        })
+        .isVisible(createTask.getBillSelect(billOption), function(result){ //Click the desired option from the 'Bill to Client' list
+            if(result.status === c.ELEMENT_FOUND)
+                browser.click(createTask.getBillSelect(billOption));
+        })
+        .assert.containsText(createTask.taskBillSelected, billOption) //Check if the selected value corresponds
+        .isVisible(createTask.taskRate, function(result){
+            if(result.status === c.ELEMENT_FOUND)
+            {
+                browser.clearValue(createTask.taskRate);
+                browser.setValue(createTask.taskRate, 5.25);
+            }
+        })
+        .isVisible(common.submitButton, function(result){ //Click the 'Save' button
+            if(result.status === c.ELEMENT_FOUND)
+                browser.click(common.submitButton);
+        })
+        .isVisible(createTask.taskPage, function(result){ //Click on the 'Tasks' section
+            if(result.status === c.ELEMENT_FOUND)
+                browser.click(createTask.taskPage);
+        })
+        .assert.containsText(common.pageTitle, pageNameTasks) // Check if the page is 'Tasks'
+        .useXpath()
+        .isVisible(tasks.nameFilter, function(result){ // Enter a search criteria
+            if(result.status === c.ELEMENT_FOUND)
+                browser.setValue(tasks.nameFilter, taskName)
+        })
+        .useCss()
+        .waitForElementVisible(tasks.getTaskName(taskName)) // Check if we see our task
```

Comment:
This does not assert if task is there. It only waits for it to appear, and it might not show up and test case would still pass.